pkg/controller/certificatesigningrequest/ipsec_csr_signing_controller.go

```diff
@@ -0,0 +1,480 @@
+
+	caConfigMapInformer := corev1informers.NewFilteredConfigMapInformer(client, env.GetAntreaNamespace(), resyncPeriod, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc}, func(listOptions *metav1.ListOptions) {
+		listOptions.FieldSelector = fields.OneTermEqualSelector("metadata.name", ipsecRootCAName).String()
+	})
+
+	configmapLister := corev1listers.NewConfigMapLister(caConfigMapInformer.GetIndexer())
+
+	c := &IPsecCSRSigningController{
+		client:                client,
+		csrInformer:           csrInformer,
+		csrLister:             csrLister,
+		csrListerSynced:       csrInformer.HasSynced,
+		configMapInformer:     caConfigMapInformer,
+		configMapLister:       configmapLister,
+		configMapListerSynced: caConfigMapInformer.HasSynced,
+		selfSignedCA:          selfSignedCA,
+		queue:                 workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(minRetryDelay, maxRetryDelay), "certificateSigningRequest"),
+		fixturesQueue:         workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(minRetryDelay, maxRetryDelay), "certificateSigningRequest"),
+	}
+
+	csrInformer.AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: c.enqueueCertificateSigningRequest,
+			UpdateFunc: func(old, cur interface{}) {
+				c.enqueueCertificateSigningRequest(cur)
+			},
+		},
+		resyncPeriod,
+	)
+
+	caConfigMapInformer.AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				c.fixturesQueue.Add(workerItemKey)
+			},
+			UpdateFunc: func(old, cur interface{}) {
+				c.fixturesQueue.Add(workerItemKey)
+			},
+			DeleteFunc: func(obj interface{}) {
+				c.fixturesQueue.Add(workerItemKey)
+			},
+		},
+		resyncPeriod,
+	)
+
+	return c
+}
+
+// Run begins watching and syncing of the IPsecCSRSigningController.
+func (c *IPsecCSRSigningController) Run(stopCh <-chan struct{}) {
+	defer c.queue.ShutDown()
+
+	klog.Infof("Starting %s", ipsecCSRSigningControllerName)
+	defer klog.Infof("Shutting down %s", ipsecCSRSigningControllerName)
+
+	go c.configMapInformer.Run(stopCh)
+	go c.csrInformer.Run(stopCh)
+
+	cacheSyncs := []cache.InformerSynced{c.csrListerSynced, c.configMapListerSynced}
+	if !cache.WaitForNamedCacheSync(ipsecCSRSigningControllerName, stopCh, cacheSyncs...) {
+		return
+	}
+	c.fixturesQueue.Add(workerItemKey)
+
+	go wait.Until(c.fixturesWorker, time.Second, stopCh)
+
+	go wait.Until(func() {
+		if err := c.watchSecretChanges(stopCh); err != nil {
+			klog.ErrorS(err, "Watch Secret error", "secret", ipsecRootCAName)
+		}
+	}, time.Minute, stopCh)
+
+	for i := 0; i < defaultWorkers; i++ {
+		go wait.Until(c.csrWorker, time.Second, stopCh)
+	}
+
+	<-stopCh
+}
+
+func (c *IPsecCSRSigningController) syncRootCertificateAndKey() error {
+	var caBytes, caKeyBytes []byte
+	caSecret, err := c.client.CoreV1().Secrets(env.GetAntreaNamespace()).Get(context.TODO(), ipsecRootCAName, metav1.GetOptions{})
```

Comment:
Why doesn't the controller create a secretInformer like configMapInformer and avoid sending both watch and get requests to apiserver? `watchSecretChanges` can be replaced by event handler as well.

Reply:
make sense